CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+    - [Background Page](#background-page)
+    - [Popup Page](#popup-page)
+    - [Activity Log Page - Tab](#activity-log-page---tab-extension-page)
+    - [Activity Log Page - Devtools](#activity-log-page---devtools-extension-page)
+  - [Core Features](#core-features)
+    - [Collecting Logs](#collecting-logs)
+      - [Live Logging](#live-logging)
+      - [Loading / saving logs](#loading--saving-logs)
+    - [Rendering Logs](#rendering-logs)
+    - [Filtering Log Entries](#filtering-log-entries)
+- [Picking an issue](#picking-an-issue)
+- [Installation](#installation)
+- [Writing and Running Tests](#writing--running-tests)
+- [Checking for Linter](#checking-for-linter)
+- [Checking for Prettier](#checking-for-prettier)
+- [Submitting a Bug or Issue or Feature Request](#submitting-a-bug-or-issue-or-feature-request)
+- [Creating a Pull Request](#creating-a-pull-request)
+  - [Writing Commit Message](#writing-commit-message)
+
+## Project Architecture
+
+Extension Activity Monitor is composed by a backend part (running in the background page) and a frontend part (composed by a browserAction Popup and an Activity Log extension page which is loaded as a tab or as a devtools panel).
+
+### Some Important Components
+
+#### Background Page
+
+The background page acts as the backend of the extension. The core part of the background page is implemented in [`ext-monitor.js`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/ext-monitor.js). It is responsible for monitoring extensions, storing activity logs, saving logs to a JSON file and loading logs from JSON file. The logs remain alive in the background unless it receives the [`clearlogs`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L106) instruction from Activity Log page(frontend part). The background page also [sends the real-time logs](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L25) to Activity Log page. The background page communicates with popup and activity log page (which are part of frontend) more often to make sure monitoring runs smoothly.
+
+#### Popup Page
+
+The popup page is one part of the frontend that communicates with the background page (backend part) to run most of its functionality. It is responsible for "start monitoring" and "stop monitoring" other installed extensions. It also helps to access the Activity Log page.
+
+#### Activity Log Page - Tab (Extension Page)
+
+The Activity Log page is another part of the frontend. It uses the MVC architecture and [web components](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component) for table view and filtering options.
+
+- The **role of Model** is to store the activity logs from background and store the filters. It also checkes if logs are matching or not matching with the applied filters.
```

Comment:
No need to repeat **role of** while explaining what every part of the activity log page MVC is responsible for.
One of the advantages of using a known terminology (like "MVC" in this case) is that you can assume that the reader knows what is MVC (or it can get that knowledge on its own) without having to restate it in the project's developers docs.

Here we can just say that "The Model class does store activity logs being rendered in the activity log page and the data representation of the log filters. The Model class does also provide a method that is being used to check if a particular log entry does match the filters."